grouper/fe/templating.py

```diff
@@ -78,3 +122,19 @@ def __init__(self, settings, deployment_name):
             "external_js": EXTERNAL_JS,
         }
         self.environment.globals.update(template_globals)
+
+    def csp_header(self):
+        # type: () -> str
+        """Return the value for the Content-Security-Policy header."""
+        policy = "frame-ancestors 'none'; form-action 'self'; default-src 'none'"
+        policy += "; img-src 'self' " + " ".join(self._cdnjs_urls(EXTERNAL_IMG))
+        policy += "; script-src 'self' " + " ".join(self._cdnjs_urls(EXTERNAL_JS))
+        policy += "; style-src 'self' " + " ".join(self._cdnjs_urls(EXTERNAL_CSS))
+        policy += "; font-src " + " ".join(self._cdnjs_urls(EXTERNAL_FONTS))
+        return policy
+
+    def _cdnjs_urls(self, resources):
+        # type: (Iterable[ExternalResource]) -> List[str]
+        """Return the URLs on the preferred CDNJS mirror for the given resources."""
+        cdnjs_prefix = cast(FrontendSettings, self.settings).cdnjs_prefix
```

Comment:
Feels wrong that the cast is necessary here... don't see a better option though

Reply:
Yeah, I feel like this is a bug in mypy. Even if I set `self.settings` explicitly in the constructor here, it still uses the type from the parent class and still produces an error without the `cast`. mypy in general doesn't deal with different method signatures on parent and child objects very well.